addons/account/tests/test_reconciliation.py

```diff
@@ -65,68 +53,72 @@ def test_statement_usd_invoice_chf_transaction_chf(self):
             {'debit': 0.0,      'credit': 10.74,    'account_id': self.diff_income_account.id},
         ])
 
     def test_statement_eur_invoice_usd_transaction_usd(self):
         customer_move_lines, supplier_move_lines = self.make_customer_and_supplier_flows(self.currency_usd_id, 50, self.bank_journal_euro, 40, 50, self.currency_usd_id)
-        self.assertRecordValues(customer_move_lines, [
-            {'debit': 0.0,      'credit': 40.0,     'amount_currency': -50, 'currency_id': self.currency_usd_id},
+        self.assertRecordValues(customer_move_lines.sorted('balance', reverse=True), [
             {'debit': 40.0,     'credit': 0.0,      'amount_currency': 50,  'currency_id': self.currency_usd_id},
+            {'debit': 0.0,      'credit': 40.0,     'amount_currency': -50, 'currency_id': self.currency_usd_id},
         ])
         exchange_lines = customer_move_lines.mapped('full_reconcile_id.exchange_move_id.line_ids')
-        self.assertRecordValues(exchange_lines, [
+        self.assertRecordValues(exchange_lines.sorted('balance', reverse=True), [
             {'debit': 7.30,     'credit': 0.0,      'account_id': self.account_rcv.id},
             {'debit': 0.0,      'credit': 7.30,     'account_id': self.diff_income_account.id},
         ])
 
-        self.assertRecordValues(supplier_move_lines, [
+        self.assertRecordValues(supplier_move_lines.sorted('balance', reverse=True), [
             {'debit': 40.0,     'credit': 0.0,      'amount_currency': 50,  'currency_id': self.currency_usd_id},
             {'debit': 0.0,      'credit': 40.0,     'amount_currency': -50, 'currency_id': self.currency_usd_id},
         ])
         exchange_lines = supplier_move_lines.mapped('full_reconcile_id.exchange_move_id.line_ids')
-        self.assertRecordValues(exchange_lines, [
-            {'debit': 0.0,      'credit': 7.30,     'account_id': self.account_rsa.id},
+        self.assertRecordValues(exchange_lines.sorted('balance', reverse=True), [
             {'debit': 7.30,     'credit': 0.0,      'account_id': self.diff_expense_account.id},
+            {'debit': 0.0,      'credit': 7.30,     'account_id': self.account_rsa.id},
         ])
 
     def test_statement_eur_invoice_usd_transaction_eur(self):
         customer_move_lines, supplier_move_lines = self.make_customer_and_supplier_flows(self.currency_usd_id, 50, self.bank_journal_euro, 40, 0.0, False)
-        self.assertRecordValues(customer_move_lines, [
-            {'debit': 0.0,      'credit': 40.0,     'amount_currency': 0.0, 'currency_id': False},
+        self.assertRecordValues(customer_move_lines.sorted('balance', reverse=True), [
             {'debit': 40.0,     'credit': 0.0,      'amount_currency': 0.0, 'currency_id': False},
+            {'debit': 0.0,      'credit': 7.3,      'amount_currency': 0.0, 'currency_id': False},
+            {'debit': 0.0,      'credit': 32.7,     'amount_currency': 0.0, 'currency_id': False},
```

Comment:
Invoice is $50 and that's worth 32.7 € (company currency), is that it?

Reply:
Your invoice:

|account | debit | credit | currency | amount_currency |
|--|--|--|--|--|
| receivable | 32.7 | 0.0 | USD | 50.0 |
| sales | 0.0 | 32.7 | USD | -50.0 |

Statement line:
journal_currency=EUR, amount=40, currency_id=USD, amount_currency=0.0

**Before this commit:**

Payment amls:
|account | debit | credit | amount_residual |
|--|--|--|--|
| bank | 40.0 | 0.0 | 40.0|
| receivable | 0.0 | 40.0 | -7.3 |

**After this commit:**

statement amls:
|account | debit | credit | amount_residual |
|--|--|--|--|--|--|
| bank | 40.0 | 0.0 | 40.0 |
| receivable | 0.0 | 32.7 | 0.0 |
| receivable | 0.0 | 7.3 | 7.3 |

The last line is an open balance.

Is it ok for you?

Reply:
I need to make something about that I guess :disappointed:

Reply:
This is only something happening in the tests because the counterpart amount is set to 40.0. Otherwise, when passing through the reconciliation widget, an open balance is generated. So, I think there is nothing to fix here.

Reply:
agreed